src/Particle/HDFWalkerInput_0_4.cpp

```diff
@@ -74,20 +70,22 @@ void HDFWalkerInput_0_4::checkOptions(xmlNodePtr cur)
         int pid = myComm->rank(), ip = 0;
         while (pid < i_info.nprocs && ip < np)
         {
-          char* h5name = new char[froot.size() + 10];
-          sprintf(h5name, "%s.p%03d", froot.c_str(), pid++);
-          FileStack.push(h5name);
-          delete[] h5name;
+          // 2 chars for '.p', 11 chars for pid, 1 char for \0
+          std::array<char, 14> h5name;
+          snprintf(h5name.data(), h5name.size(), ".p%03d", pid++);
+          auto fname = std::filesystem::path(froot).concat(h5name.data());
+          FileStack.push(std::move(fname));
           pid += nprocs_now;
         }
       }
       else
       {
-        int pid      = myComm->rank() % i_info.nprocs;
-        char* h5name = new char[froot.size() + 10];
-        sprintf(h5name, "%s.p%03d", froot.c_str(), pid);
-        FileStack.push(h5name);
-        delete[] h5name;
+        int pid = myComm->rank() % i_info.nprocs;
+        // 2 chars for '.p', 11 chars for pid, 1 char for \0
```

Comment:
The source of the "11 characters" is a little confusing at first glance because the format string is "%03d".  It looks like it should be zero-padded with a width of 3.  However, the width specifier is a minimum width - printf formatting expands the field to accommodate larger values.  The type of pid is an int, and so the maximum number of characters need to represent the int with the longest string is 11 (For instance, -2**31)

Reply:
Yes, I made that mistake in the first commit and am thankful that GCC caught it. clang didn't generate any warnings. 

The `+10` buffer currently used could also potentially overflow, though I suspect other things would also break with 10,000,000+ processes.